src/RNTupleReader.cc

```diff
@@ -176,6 +178,10 @@ std::unique_ptr<ROOTFrameData> RNTupleReader::readEntry(const std::string& categ
 #endif
 
   for (size_t i = 0; i < m_collectionInfo[category].id.size(); ++i) {
+    if (!collsToRead.empty() &&
+        std::ranges::find(collsToRead, m_collectionInfo[category].name[i]) == collsToRead.end()) {
+      continue;
+    }
```

Comment:
```suggestion
    if (std::ranges::find(collsToRead, m_collectionInfo[category].name[i]) == collsToRead.end()) {
      continue;
    }
```
Probably nothing is gained by checking if `find` checks or does a for loop over its elements? Same below

Reply:
I just edited; I just meant the first empty check as it's deleted in the suggestion

Reply:
Ah, sorry then I completely misunderstood. The check is negated, so we only want to do the `find` if there are collections to check. If we remove the check, we will stop reading any collections (because if `collsToRead` is empty then `find` will never find a name in it).

Reply:
I think my suggestion is still incorrect because how it is now if `collsToRead` is empty then the first check will always fail and will always continue. So it should be like this, only continue if at least one collection is passed to be read and it is not found there:
```
    if (!collsToRead.empty()) {
      if (std::ranges::find(collsToRead, m_collectionInfo[category].name[i]) == collsToRead.end()) {
        continue;
      }
    }
```
then the existing calls will just pass an empty `collsToRead` and the `continue` will not trigger.

Reply:
Unless I am missing something the nested `if` and the current implementation with the `&&` are equivalent?

The main question this discussion has triggered for me is whether the `continue` should actually be an exception, because arguably, if I put a collection name into the `collsToRead`, I expect it to be in the `Frame` I get in the end.

Reply:
Ah yes, it's correct. Probably an exception is better than simply not reading it.

Reply:
Now throwing an exception in case a non-existent collection is requested.